### DISCUSSION TITLE
[API] Endpoints to get list of supported Source/Destination Definitions and source/Destination configuration schema

### DISCUSSION THREAD

**Discussion by ameyabapat-bsft** [airbytehq/airbyte#33140] [Connector Ideas and Features]
### What area the feature impact?

Airbyte Platform

### Relevant Information

To create a dynamic UI to support different sources/destinations on top of Airbyte APIs, 
1. it is required to know the supported list of source and destination definitions.
2. Schema for a particular source or destination. 
If we use airbyte APIs, currently we have to harcode/persist the definition Ids and schema for desired sources and destination. It is not possible to automate the flow.

Similar APIs are available in config APIs. 
Ex 
https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/source_definitions/list_for_workspace
https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/source_definition_specifications/get

Migrating completely from config APIs to Airbyte API is difficult due to such missing endpoints. https://docs.airbyte.com/api-documentation
Such endpoints could truly make it possible to support any source/destination configuration based on API.

**Comment by marcosmarxm** [comment]
Thanks for the suggestion @ameyabapat-bsft I moved the issue to a discussion to be easier to get feedback from other contributors.

[a comment by an automated account is not shown]

**Comment by ameyabapat-bsft** [comment]
@airbyte-support-bot can I get some feedback from airbyte on this matter?
